src/File.hack

```diff
@@ -9,11 +9,14 @@
 
 namespace Facebook\HHAST;
 
+use namespace HH\Lib\{C, Str};
+
 final class File {
   private function __construct(
     private string $path,
     private string $contents,
     private bool $isDirty,
+    private string $extension,
```

Comment:
Why not just compute the extension from the path inside `isHackFile()` (the only place where it's used), instead of passing it in? It would save a bit of code, and prevent people from accidentally introducing bugs if they pass in an inconsistent combination of path + extension somehow in the future.

Reply:
That is a good call indeed